### PR TITLE
WT-6759 initial fetch client implementation

### PR DESCRIPTION
**Pull request by pfilippi24**
Implementation of fetch clients
Moved posted time health check from HealthCheck class to the respective runners
Changes ServerRunner to ConnectorRunner and provided means to choose between webhook and fetch client implementations.

Did testing using the folder based connector

### CODE REVIEW

File: src/main/java/io/wisetime/connector/fetch_client/FetchClient.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                PostResult result = connector.postTime(null, timeGroup);
+                try {
+                  // TRANSIENT_FAILURE doesn't need to be handled.
+                  // simply sending no update is considered a transient failure after a certain timeout
+                  switch (result) {
+                    case SUCCESS:
+                      apiClient.updatePostedTimeStatus(new TimeGroupStatus()
+                          .status(TimeGroupStatus.StatusEnum.SUCCESS)
+                          .timeGroupId(timeGroup.getGroupId())
+                          .fetchClientId(fetchClientId));
```
Comment (Galya-IT) [inline src/main/java/io/wisetime/connector/fetch_client/FetchClient.java]: The GUI app should show what happens behind the scenes, so we need meaningful `log.info` for any of the result cases. Also there are events that should be triggered on success, please check how they are used in other places - `TAGS_UPSERTED` and `TIME_GROUP_POSTED.

Comment (pfilippi24) [reply src/main/java/io/wisetime/connector/fetch_client/FetchClient.java]: Thanks, will add it

File: src/main/java/io/wisetime/connector/fetch_client/FetchClient.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    this.lastSuccessfulRun = new AtomicReference<>(DateTime.now());
+  }
+
+  @Override
+  public void run() {
+    while (!Thread.currentThread().isInterrupted()) {
+      try {
+        apiClient.fetchTimeGroups(fetchClientId, limit).forEach(timeGroup ->
+            postTimeExecutor.submit(() -> {
+              if (!timeGroupIdStore.alreadySeen(timeGroup.getGroupId())) {
```
Comment (Galya-IT) [inline src/main/java/io/wisetime/connector/fetch_client/FetchClient.java]: Is there any scenario where a `TimeGroup` is actually left "seen" in the DB? It looks like it gets deleted in every case - success, failure, exception.

**Comment by vyshane** [comment]
@pfilippi24 I'm closing this PR, please reopen against Stash when ready.

**Pull request closed without merging**